packages/itertools/test.pony

```diff
@@ -0,0 +1,232 @@
+    let expected5 = [as USize: 14, 27, 39]
+
+    let actual1 = Array[String]
+    let actual2 = Array[U32]
+    let actual3 = Array[F32]
+    let actual4 = Array[I32]
+    let actual5 = Array[USize]
+
+    for (a1, a2, a3, a4, a5) in
+      Zip5[String, U32, F32, I32, USize](input1.values(), input2.values(),
+      input3.values(), input4.values(), input5.values()) do
+      actual1.push(a1)
+      actual2.push(a2)
+      actual3.push(a3)
+      actual4.push(a4)
+      actual5.push(a5)
+    end
+
+    h.assert_array_eq[String](expected1, actual1)
+
+    // Skipping for now because of issue 563
+    // (https://github.com/ponylang/ponyc/issues/563)
+    // h.assert_array_eq[U32](expected2, actual2)
+    // h.assert_array_eq[F32](expected3, actual3)
+    // h.assert_array_eq[I32](expected4, actual4)
+    // h.assert_array_eq[USize](expected5, actual5)
+
+class iso _TestRepeat is UnitTest
+  fun name(): String => "itertools/Repeat"
+
+  fun apply(h: TestHelper) =>
+    let input = "a"
+    let expected = ["a", "a", "a"]
+    var actual = Array[String]
+
+    let repeater = Repeat[String]("a")
+
+    actual.push(repeater.next()) // 1
+    actual.push(repeater.next()) // 2
+    actual.push(repeater.next()) // 3 ... Repeater!
+
+    h.assert_array_eq[String](expected, actual)
+
+class iso _TestTake is UnitTest
+  fun name(): String => "itertools/Take"
+
+  fun apply(h: TestHelper) =>
+    let take: USize = 3
+    let input = ["a", "b", "c", "d", "e"]
+    let expected = ["a", "b", "c"]
+    var actual = Array[String]
+
+    for x in Take[String](input.values(), take) do
+      actual.push(x)
+    end
+
+    h.assert_array_eq[String](expected, actual)
+
+class iso _TestCycle is UnitTest
+  fun name(): String => "itertools/Cycle"
+
+  fun apply(h: TestHelper) =>
+    let input = ["a", "b", "c"]
+    let expected = ["a", "b", "c", "a", "b", "c", "a", "b", "c", "a"]
+    var actual: Array[String] = Array[String]
+
+    let cycle = Cycle[String](input.values())
+    try
+      actual.push(cycle.next()) // 1 "a"
+      actual.push(cycle.next()) // 2 "b"
+      actual.push(cycle.next()) // 3 "c"
+      actual.push(cycle.next()) // 4 "a" <= start first cached cycle
+      actual.push(cycle.next()) // 5 "b"
+      actual.push(cycle.next()) // 6 "c"
+      actual.push(cycle.next()) // 7 "a" <= start second cached cycle
+      actual.push(cycle.next()) // 8 "b"
+      actual.push(cycle.next()) // 9 "c"
+      actual.push(cycle.next()) // 10 "a" <= start third cached cycle
+    else
+      h.fail()
+    end
+    
+    h.assert_array_eq[String](expected, actual)
+
+class iso _TestMapFn is UnitTest
+  fun name(): String => "itertools/MapFn"
+
+  fun apply(h: TestHelper) =>
+    let input = ["a", "b", "c"]
+    let expected = ["ab", "bb", "cb"]
+    let actual = Array[String]
+
+    for x in MapFn[String, String](input.values(),
+                                   lambda (x: String): String =>
+                                   x + "b" end) do
```

Comment:
I'm not sure what we should use for the `stdlib` formatting rules for lambdas, but I personally don't like this formatting very much.

I think I'd rather see something like this:
```pony
    for x in MapFn[String, String](input.values(),
                                   lambda (x: String): String =>
                                     x + "b"
                                   end) do
      actual.push(x)
    end
```

Or maybe like this:
```pony
    for x in MapFn[String, String](
      input.values(),
      lambda (x: String): String => x + "b" end
    ) do
      actual.push(x)
    end
```

Or perhaps it is truly best to just put the lambda on its own line:
```pony

    let fn = lambda (x: String): String => x + "b" end
    for x in MapFn[String, String](input.values(), fn) do
      actual.push(x)
    end
```
```pony